bots/notify/src/main/java/org/openjdk/skara/bots/notify/PullRequestWorkItem.java

```diff
@@ -277,6 +286,12 @@ public Collection<WorkItem> run(Path scratchPath) {
             if (!storedState.get().state().equals(state.state())) {
                 notifyStateChange(storedState.get().state(), scratchPath);
             }
+            var storedTargetBranch = storedState.get().targetBranch();
+            if (storedTargetBranch == null || !storedTargetBranch.equals(state.targetBranch())) {
```

Comment:
I think we need to flip this check. If `storedTargetBranch` is null and we for some reason get a null from `state.targetBranch()`, then this will keep notifying even though there is no change. If we instead flip it to
```
state.targetBranch() != null && !state.targetBranch().equals(storedTargetBranch)
```
then we will only notify if the updated branch is not null. It may never be needed, but it's a bit more defensive against unexpected data being returned from the forge.

Reply:
Yes, it's a good point. Thanks!